python/heterocl/tvm/schedule.py

```diff
@@ -256,100 +256,60 @@ def reuse_at(self, target, parent, axis, name):
     def partition(self, target, partition_type, dim, factor):
         return _api_internal._SchedulePartition(self, target, dim, factor, partition_type)
 
-    def join(self, target, dst, src, type=_expr.IO.FIFO, depth=1):
-        """ join multiple writes to target tensor """
-        return _api_internal._ScheduleJoin(self, target, src, dst, type, depth)
-
-    def to(self, tensor, dst, src, axis=0,
-           type=_expr.IO.DMA, depth=1, local_buffer=True):
-        """ Stream data to devices or on-chip module
+    def to(self, api_mode, target, dst, src, axis=0, 
```

Comment:
Instead of having a single `.to` and using `if` and `elif`, can we divide it into several functions? It would be clearer. For instance, `.in_stage_move` for `ScheduleInStageMove`.